### ISSUE TITLE
baseline-source CSS property

### ISSUE THREAD

**Issue by bfgeek** [WebKit/standards-positions#118]
## Request for position on an emerging web specification

* WebKittens who can provide input:

### Information about the specification

* Title: CSS Inline 3
* URL: https://drafts.csswg.org/css-inline-3/#baseline-source
* GitHub repository: https://github.com/w3c/csswg-drafts

### Design reviews and vendor positions

* TAG Design Review:
* Mozilla standards-positions issue: https://github.com/mozilla/standards-positions/issues/727

### Bugs tracking this feature

* WebKit Bugzilla: https://bugs.webkit.org/show_bug.cgi?id=249094
* Radar:

### Anything else we need to know

The "baseline-source" properties allows web developers to specify if an inline-level box should use the "first" or "last" baseline for alignment within an linebox.

Today the default behaviour is confusing for web developers. Consider:

```
test <div style="display: inline-block;">line1<br>line2</div>
test <div style="display: inline-flex;">line1<br>line2</div>
```

The "inline-block" will align to the last baseline, and the "inline-flex" will align to the first baseline.

`baseline-source: auto` is the existing (confusing) behaviour.

Web developers can specify `baseline-source: first` or `baseline-source: last` to directly determine how they want these boxes to align within a line-box.

See demo here: https://twitter.com/bfgeek/status/1612518098611089408

**Comment by annevk** [comment]
cc @nt1m @jensimmons @litherum

**Comment by nt1m** [comment]
@alanbaradlay @sgill26 can comment better on the layout bits, first/last baseline alignment was implemented for grid/flexbox, so I assume this is mostly a matter of adjusting the existing `vertical-align` code (not sure how hard it is).

I do notice the spec defines `baseline-source` as one of the longhands for `vertical-align` (along with `baseline-shift` and `alignment-baseline`), which is something I see Chrome doesn't implement right now. That bit is slightly time consuming with not a lot of return, so it would be nice if someone could prototype this and see if there are any issues with the syntax (e.g. syntax is too expressive, meaning there's a lot more work hidden, etc.). @bfgeek Do you have some insights about this?

**Comment by bfgeek** [comment]
That's right - we'll likely do the `vertical-align` mapping separately, and there is a bit of complexity hidden behind the mapping, and a fair amount of new behaviour ( e.g. `vertical-align: text-top 10px` would be valid where it wasn't before, and maps to properties that previously only applied to SVG ).

It isn't as simple as adjusting the parser, and mapping to those properties.

Ian
